polkadot/runtime/parachains/src/inclusion/mod.rs

```diff
@@ -1193,11 +1232,15 @@ impl AcceptanceCheckErr {
 	}
 }
 
-impl<T: Config> OnQueueChanged<AggregateMessageOrigin> for Pallet<T> {
+impl<T: Config> OnQueueChanged<T::AggregateMessageOrigin> for Pallet<T> {
 	// Write back the remaining queue capacity into `relay_dispatch_queue_remaining_capacity`.
-	fn on_queue_changed(origin: AggregateMessageOrigin, fp: QueueFootprint) {
-		let para = match origin {
-			AggregateMessageOrigin::Ump(UmpQueueId::Para(p)) => p,
+	fn on_queue_changed(origin: T::AggregateMessageOrigin, fp: QueueFootprint) {
+		let para: ParaId = match origin.try_into() {
+			Ok(id) => id,
+			Err(_) => {
+				log::warn!(target: LOG_TARGET, "on_queue_changed: non paraId origin, ignoring");
```

Comment:
I don't think it has to be a warning since this is a valid possibility if we have multiple types of origins. Maybe at most a `debug` log.